src/Microsoft.ML.TimeSeries/RootCauseAnalyzer.cs

```diff
@@ -702,7 +702,7 @@ private static bool ContainsAll(Dictionary<string, Object> bigDictionary, Dictio
 
         private bool IsAggregationDimension(Object val, Object aggSymbol)
         {
-            return Convert.ToString(val).Equals(aggSymbol);
```

Comment:
This line is the root cause of the issue. Converting one parameter into string makes the value type different if the object isn't string.
And I replace all equality of object by Object.Equals() to make sure it works as expected when any object is null.